resources/run_adl.sh

```diff
@@ -0,0 +1,158 @@
+  distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+  if [ "$distro" = "\"Red Hat Enterprise Linux\"" ] ||
+[ "$distro" = "\"Red Hat Enterprise Linux Server\"" ]; then
+    export VARIANT=rhel7
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_REDHAT
+  elif [ "$distro" = "\"Ubuntu\"" ]; then
+    export VARIANT=ubuntu1804  
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_UBUNTU
+  else
+    echo ${distro} not supported
+    exit 1
+  fi
+elif [ $OS = "Darwin" ]; then
+  MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_MAC
+else
+  echo $(uname) not supported
+  exit 1
+fi
+
+check_mongod
+if [[ $? -ne 0 ]]; then
+  # Install and start mongod
+  if [ $OS = "Linux" ]; then
+    curl -O $MONGO_DOWNLOAD_BASE/linux/$MONGO_DOWNLOAD_LINK
+  else
+    curl -O $MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_LINK
+  fi
+  
+  tar zxvf $MONGO_DOWNLOAD_LINK
+  
+  mkdir -p $MONGO_DB_PATH
+  mkdir -p $MONGO_LOGS_PATH
+
+  # Note: ADL has a storage.json file that generates configs for us. The mongodb source is on port 28017 so we use that here.
+  ${MONGO_DOWNLOAD_LINK:0:$((${#MONGO_DOWNLOAD_LINK} - 4))}/bin/mongod --port $MONGOD_PORT --dbpath $MONGO_DB_PATH --logpath $MONGO_LOGS_PATH/test.log --fork
+fi
+
+check_mongohoused
+if [[ $? -ne 0 ]]; then
+  # Install and start mongohoused 
+  git config --global url.git@github.com:.insteadOf https://github.com/
+  # Clone the mongohouse repo
+  if [ ! -d "$MONGOHOUSE_DIR" ]; then
+    git clone $MONGOHOUSE_URI
+  fi
+  cd $MONGOHOUSE_DIR
+  git pull $MONGOHOUSE_URI
+  
+  OS_LOWERCASE=$(echo $OS | tr '[:upper:]' '[:lower:]')
+  curl -O "https://dl.google.com/go/${GOVERSION}.${OS_LOWERCASE}-amd64.tar.gz"
+  mkdir $GO_DIR
+  tar -C $GO_DIR -xzf ${GOVERSION}.${OS_LOWERCASE}-amd64.tar.gz
+  export PATH=$GO_DIR/go/bin:$PATH
+  # May have been set from previous install
+  unset GOROOT
+  
+  go mod vendor
+
+  # Download external dependencies
+  go run cmd/buildscript/build.go tools:download:mqlrun
+  go run cmd/buildscript/build.go tools:download:mongosql
+  
+  # Set relevant environment variables
+  export GOPRIVATE=github.com/10gen
+  export MONGOHOUSE_ENVIRONMENT="local"
+  export MONGOHOUSE_MQLRUN="$(pwd)/artifacts/mqlrun"
+  export LIBRARY_PATH="$(pwd)/artifacts"
+  
+  # Load tenant config into mongodb
+  # TODO - we should probably make a different tenant config than the one that exists at
+  #  ./testdata/config/mongodb_local/tenant-config.json since that one includes a ton of extra datasources
+  #  we not only don't need but likely don't want. We'll probably want a tenant config that has a wildcard
+  #  so we can have any databases and collections in the local mongod.
+  #  - We'll probably want the test runner, as opposed to this script, to be able to say what data should get
+  #    restored.
+  go run cmd/buildscript/build.go init:mongodb-tenant 
+  
+  # Start mongohoused with appropriate config
+  go run -tags mongosql ./cmd/mongohoused/mongohoused.go --config ./testdata/config/mongodb_local/frontend-agent-backend.yaml
```

Comment:
One thing to determine, and this may also depend on how we are running this script, is how we want the script to end.  The last line is the `go run` command so the script will has to continue running, and the output is to stdout.  Another alternative would be to use `nohup` and send the output to a log file.

Reply:
Sounds good.  Made the change to use `nohup` and sends the output to a log file.